### PR TITLE
added example for sorting keys

### PR DESCRIPTION
**Pull request by sekyHC**
We needed a way to sort keys and lost a day to find out how to do it. That is why we are contributing back to this awesome library with an example

### CODE REVIEW

File: example/sortkeys/sortkeys.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#include <rapidjson/prettywriter.h>
+#include <rapidjson/stringbuffer.h>
+
+#include <algorithm>
+#include <iostream>
+
+using namespace rapidjson;
+using namespace std;
+
+void printIt(Document &doc)
```
Comment (miloyip) [inline example/sortkeys/sortkeys.cpp]: `const Value&` is sufficient

File: example/sortkeys/sortkeys.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
+#define RAPIDJSON_HAS_STDSTRING 1
```
Comment (miloyip) [inline example/sortkeys/sortkeys.cpp]: I think it may be better to have this example without using `std::string`

**Comment by miloyip** [comment]
It still cannot compile. Due to not supporting pre-C++11?

**Comment by sekyHC** [comment]
I am compiling it as per instructions here: https://github.com/Tencent/rapidjson#installation

output:
```
 ~/s/rapidjson   sort_by_name_example  build  make                                                                                               10:30
[  2%] Built target simplepullreader
[  5%] Built target tutorial
[  7%] Built target jsonx
[ 10%] Built target pretty
Scanning dependencies of target sortkeys
[ 11%] Building CXX object example/CMakeFiles/sortkeys.dir/sortkeys/sortkeys.cpp.o
[ 12%] Linking CXX executable ../bin/sortkeys
[ 12%] Built target sortkeys
[ 15%] Built target condense
[ 17%] Built target simpledom
[ 20%] Built target messagereader
[ 23%] Built target simplewriter
[ 25%] Built target filterkeydom
[ 28%] Built target filterkey
[ 30%] Built target lookaheadparser
[ 33%] Built target capitalize
[ 35%] Built target parsebyparts
[ 38%] Built target prettyauto
[ 41%] Built target schemavalidator
[ 43%] Built target simplereader
[ 46%] Built target serialize
[ 46%] Built target examples
[ 50%] Built target archivertest
[ 52%] Built target namespacetest
[ 55%] Built target gtest
[ 57%] Built target gtest_main
[ 92%] Built target unittest
[100%] Built target perftest
[100%] Built target tests
```

**Pull request merged**